services/src/test/scala/cromwell/pipeline/service/RunServiceTestImpl.scala

```diff
@@ -0,0 +1,40 @@
+package cromwell.pipeline.service
+import cromwell.pipeline.datastorage.dto.{ Run, RunCreateRequest, RunUpdateRequest }
+import cromwell.pipeline.model.wrapper.{ RunId, UserId }
+
+import scala.concurrent.Future
+
+class RunServiceTestImpl(runs: Seq[Run], testMode: TestMode) extends RunService {
+
+  override def addRun(runCreateRequest: RunCreateRequest, userId: UserId): Future[RunId] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(runs.head.runId)
```

Comment:
Same problem as with `UserServiceTestImpl.addUser`

Reply:
That one is trickier. Should a `RunId` be created by random Id generator inside the test method then, to be sure that smth will be returned anyway?

Reply:
Apparently so

Reply:
Done